includes/admin/class-wc-rest-payments-settings-controller.php

```diff
@@ -513,7 +513,6 @@ public function get_settings(): WP_REST_Response {
 				'is_card_present_eligible'               => $this->wcpay_gateway->is_card_present_eligible() && isset( WC()->payment_gateways()->get_available_payment_gateways()['cod'] ),
 				'is_woopay_enabled'                      => 'yes' === $this->wcpay_gateway->get_option( 'platform_checkout' ),
 				'show_woopay_incompatibility_notice'     => get_option( 'woopay_invalid_extension_found', false ),
-				'show_express_checkout_incompatibility_notice' => $this->should_show_express_checkout_incompatibility_notice(),
```

Comment:
> Note that this PR only ensures that the notice will no longer appear. I am leaving the client-side code in place, because we plan to revisit this functionality and hopefully will be able to reuse those components created in https://github.com/Automattic/woocommerce-payments/pull/8339.

I get it that the client-side can be repurposes, but I don't think I agree that we should keep its dead code in the codebase.
The `getShowExpressCheckoutIncompatibilityNotice()` selector, `useExpressCheckoutShowIncompatibilityNotice()` hook, and all their usages would just become obsolete.
Pulling them back would just be a matter of reverting these changes + modifying their functionality.

Keeping the code would introduce unnecessary clutter and create potential maintenance issues.

I understand that it's nice that the front-end code could keep living, but the uncertainty of when it would be re-introduced + the simplicity of eliminating it at this time + the uncertainty of whether it will keep working properly once we try to re-utilize it makes me lean towards eliminating all the dead code with this PR.